internal/server/handlers.go

```diff
@@ -623,20 +623,17 @@ func (a *API) UpdateIdentityInfoFromProvider(c *gin.Context) error {
 	// get current identity provider groups
 	info, err := oidc.GetUserInfo(providerUser)
 	if err != nil {
-		if errors.Is(err, internal.ErrForbidden) {
-			err := access.DeleteAllIdentityAccessKeys(c)
-			if err != nil {
-				logging.S.Errorf("failed to revoke invalid user session: %s", err)
-			}
-
-			deleteAuthCookie(c)
-		}
-
 		if errors.Is(err, context.DeadlineExceeded) {
 			return fmt.Errorf("%w: %s", internal.ErrBadGateway, err.Error())
 		}
 
-		return fmt.Errorf("update user info: %w", err)
+		if nestedErr := access.DeleteAllIdentityAccessKeys(c); nestedErr != nil {
```

Comment:
Err on the side of caution on update errors. Revoke sessions when something goes wrong.